### DISCUSSION TITLE
sending arbitrary uci commands

### DISCUSSION THREAD

**Discussion by pallabbasu** [niklasf/python-chess#938] [Q&A]
Hi,

I want to test my own fairy variant in fairy-stockfish and python-chess.

I want to send a few uci command to fairy-stockfish, without using chess.board(), as I do not wish any legality checking, also I am using symbols other than "pnbrqk".  I simply wish to send uci commands to fairy-stockfish (or any uci engines). 

Is there an easy way to do that in python-chess. 

Say I want to send the following commands to begin with:

setoption name UCI_variant value myvariant
position startpos move e2e4
go movetime 3000

And get back the best-move. 

best,
Pallab

**Comment by niklasf** [comment]
Hi. Implementing a custom command is reasonably straight-forward: https://github.com/niklasf/python-chess/blob/553506719ecc2535ca1b06962792186681307624/chess/engine.py#L1435-L1447

However much of the machinery to parse moves etc. is useless in this particular use case, so I am not sure if you gain much from using python-chess.
